pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -278,14 +278,20 @@ def should_reset_tensors(self, fx: str) -> bool:
             is_first_batch = bool(self._batch_idx) + self._split_idx == 0
         return is_different_fx and is_first_batch
 
-    def reset(self, metrics: Optional[bool] = None) -> None:
-        if self.trainer.sanity_checking:
+    def reset_metrics(self) -> None:
```

Comment:
Do not extract this if it's not going to be called from the outside.

I think it'd be better having two functions, this one and `def reset_results(metrics)`

to avoid the `trainer_metrics` flag

Reply:
I think it can be called from outside, let's say someone wants to reset their metrics after every epoch, then they can just use this in one of their hooks. Right now it resets them at every run.

Reply:
That's why I suggest keeping it public but removing the `trainer_metrics` flag